waiter/src/waiter/token.clj

```diff
@@ -205,34 +206,35 @@
              (utils/exception->json-response e)))
     :post (try
             (let [authenticated-user (get req :authorization/user)
-                  {:strs [token run-as-user] :as service-description-template} (json/read-str (slurp (:body req)))
-                  existing-service-description-template (sd/token->service-description-template kv-store token :error-on-missing false)
-                  owner (or (get service-description-template "owner")
-                            (get existing-service-description-template "owner")
+                  {:strs [token run-as-user] :as new-token-description} (json/read-str (slurp (:body req)))
+                  {existing-token-metadata :token-metadata} (sd/token->token-description kv-store token)
```

Comment:
One saving here: we know clearly that we do not care about the contents of the existing token since we do not extract `:service-description-template`.